indexer/positional_index.py

```diff
@@ -96,7 +96,52 @@ def sortPostingsLists(self):
     def phraseQuery(self, phrase):
         tokens = self.tokenizer(StringIO(phrase))
         postingsLists = [self.dictionary[token] for token in tokens]
-        raise NotImplementedError
+        setOfDocs = list()
+        answer = set()
+        for i in range(len(postingsLists) - 1):
+            plist = postingsLists[i: i+2]
+            setOfDocs.append(set(self.positionalIntersect(plist[0], plist[1])))
+
+        if setOfDocs != []:
+            answer = set(setOfDocs[0])
+            for s in setOfDocs:
+                answer = answer & s
+        return answer
+
+
+    def positionalIntersect(self, posting1, posting2):
+        answer = []
+
+        list1 = iter(posting1.postings)
+        list2 = iter(posting2.postings)
+
+        for p1, p2 in zip(list1, list2):
```

Comment:
This probably doesn't do what you want it to do. When you do `zip(list1, list2)`, you are creating new iterators that are independent of both `list1` and `list2`. And since you actually don't access the result of the `next` function, all of those next calls are redundant and do nothing. Moreover, `p1` and `p2` are normal iterations variables capped at the smallest list of both, so it doesn't follow the intersection algorithm outlined in the book.